lib/hanami/providers/db/sql_adapter.rb

```diff
@@ -6,10 +6,27 @@ class DB < Hanami::Provider::Source
       # @api public
       # @since 2.2.0
       class SQLAdapter < Adapter
+        # @api public
+        # @since 2.2.0
+        setting :connection_options, mutable: true
+
         # @api public
         # @since 2.2.0
         setting :extensions, mutable: true
 
+        # @api public
+        # @since 2.2.0
+        def connection_options
+          config.connection_options ||= {}
+        end
+
+        # @api public
+        # @since 2.2.0
+        def options(**options)
+          options.delete(:extensions)
```

Comment:
There are two ways to go here, and I believe it's better for long-term maintainability to not allow `options` to be a back-door for `extensions` since that has a definite public api.